docs/spec/spec.md

```diff
@@ -966,6 +966,8 @@ In Ballerina, `distinct` `service` objects can be used to define GraphQL interfa
 
 Non-distinct `service` objects and `service` classes can not be used to define or implement GraphQL interfaces.
 
+>**Note**: Recognition of `service` `object`s and `service` `class`es as GraphQL interfaces and implementations relies on their definition within the same module as the GraphQL service.
```

Comment:
On second thoughts, this seems unclear. Shall we reword this? IMO, the spec (or any technical documentation for that matter) should be easy to read and understand. Using excessive adjectives and complex sentences (generated by generative AIs) beats that purpose.

```suggestion
>**Note**: In order to be recognized as GraphQL objects or interfaces, the Ballerina `service` `object`s and `service` `class`es should be defined within the same module as the GraphQL service.
```